internal/handler/dataBinder.go

```diff
@@ -3,19 +3,20 @@ package handler
 import (
 	"fmt"
 
-	"github.com/dhis2-sre/im-manager/internal/apperror"
+	"github.com/dhis2-sre/im-manager/internal/errdef"
+
 	"github.com/gin-gonic/gin"
 )
 
 func DataBinder(c *gin.Context, req interface{}) error {
 	if c.ContentType() != "application/json" && c.ContentType() != "multipart/form-data" {
 		reason := fmt.Sprintf("%s only accepts content of type application/json or multipart/form-data", c.FullPath())
-		return apperror.NewUnsupportedMediaType(reason)
+		return errdef.NewUnsupportedMediaType(reason)
```

Comment:
I would follow the [im-user internal/handler/dataBinder.go](https://github.com/dhis2-sre/im-user/pull/16/files#diff-81b269c1e73a24de897a1f4e7831d0428479c171e871dfe1f9489f2deae6c4f5) and the guideline in https://github.com/dhis2-sre/im-user/pull/16

`errdef` to me should have HTTP independent errors. The HTTP error middleware than turns these into appropriate HTTP status codes.